packages/core/src/Site/index.js

```diff
@@ -1300,11 +1303,67 @@ class Site {
 
           return publish(basePath, options);
         })
-        .then(resolve)
+        .then(() => {
+          const git = simpleGit({ baseDir: process.cwd() });
+          const options = {};
+          options.remote = defaultDeployConfig.remote;
+          options.repo = this.siteConfig.deploy.repo || defaultDeployConfig.repo;
+          options.branch = this.siteConfig.deploy.branch || defaultDeployConfig.branch;
+          return Site.getDeploymentUrl(git, options);
+        })
+        .then(depUrl => (depUrl != null ? resolve(`Deployed at ${depUrl}!`) : resolve('Deployed!')))
         .catch(reject);
     });
   }
 
+  /**
+   * Gets the url where the website is deployed at.
+   */
+  static getDeploymentUrl(git, options) {
+    const HTTPS_PART = 'https://';
+    const SSH_PART = 'git@github.com:';
+    const GITHUB_IO_PART = 'github.io';
+
+    // https://<name|org name>.github.io/<repo name>/
+    function constructGhPagesUrl(remoteUrl) {
+      if (remoteUrl.includes(HTTPS_PART)) {
+        // https://github.com/<name|org>/<repo>.git (HTTPS)
+        const parts = remoteUrl.split('/');
+        const repoName = parts[parts.length - 1].toLowerCase();
+        const name = parts[parts.length - 2].toLowerCase();
+        return `https://${name}.${GITHUB_IO_PART}/${repoName}`;
+      } else if (remoteUrl.includes(SSH_PART)) {
+        // git@github.com:<name|org>/<repo>.git (SSH)
+        const parts = remoteUrl.split('/');
+        const repoName = parts[parts.length - 1].toLowerCase();
+        const name = (parts[0].split(':'))[0];
+        return `https://${name}.${GITHUB_IO_PART}/${repoName}`;
+      }
+      throw new Error(`Unknown remote url ${remoteUrl}`);
```

Comment:
Since the message here isn't used, let's just return `null` directly for use in the above.
```
depUrl => (depUrl != null ? resolve(`Deployed at ${depUrl}!`) : resolve('Deployed!'))
```

Could remove the `catch(..)` below as well